### PR TITLE
Task2b

### PR DESCRIPTION
**Pull request by danicatt**

### CODE REVIEW

File: src/Collections/PropertyCollection.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $this->propertyCheck($properties);
+        $this->allProperties = $properties;
+    }
+
+    private function propertyCheck($properties)
+    {
+        foreach ($properties as $property) {
+            if (!($property instanceof PropertyEntity)) {
+                throw new \Exception('Array must only contain instances of PropertyEntity');
```
Comment (Sen-i) [inline src/Collections/PropertyCollection.php:24]: What does the \ do here ?
can it be 
```suggestion
                throw new Exception('Array must only contain instances of PropertyEntity');
```

Comment (Sen-i) [reply src/Collections/PropertyCollection.php:24]: ohh it is related to namesapce!?

File: src/Hydrators/PropertyHydrator.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+
+namespace ArmadilloEstates\Hydrators;
+
+
+use ArmadilloEstates\Entities\PropertyEntity;
+
+class PropertyHydrator
+{
+    private $database;
```
Comment (Sen-i) [inline src/Hydrators/PropertyHydrator.php:13]: whiteSpace
```suggestion
   private $database;
```

File: src/Entities/PropertyEntity.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace ArmadilloEstates\Entities;
+
+
```
Comment (Sen-i) [inline src/Entities/PropertyEntity.php:5]:
```suggestion

```
WhiteSpace

File: src/Hydrators/PropertyHydrator.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+
```
Comment (Sen-i) [inline src/Hydrators/PropertyHydrator.php:3]:
```suggestion

```
WhiteSpace

File: src/Hydrators/PropertyHydrator.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+namespace ArmadilloEstates\Hydrators;
+
+
```
Comment (Sen-i) [inline src/Hydrators/PropertyHydrator.php:6]:
```suggestion

```
WhiteSpace

File: src/Collections/PropertyCollection.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+
```
Comment (Sen-i) [inline src/Collections/PropertyCollection.php:3]:
```suggestion

```
WhiteSpace

File: src/Collections/PropertyCollection.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+namespace ArmadilloEstates\Collections;
+
+
```
Comment (Sen-i) [inline src/Collections/PropertyCollection.php:6]:
```suggestion

```
WhiteSpace

**Review by Sen-i — CHANGES_REQUESTED** [review]
Are the interfaces not implemented for a reason?

File: src/Hydrators/PropertyHydrator.php [hunk 8]
```diff
@@ -15,11 +17,11 @@ public function __construct(\PDO $database)
         $this->database = $database;
     }
 
-    public function getAllBasicProperties()
+    public function getAllBasicProperties(): IPropertyCollection
     {
         $query = $this->database->query("SELECT `Address1`, `Town`, `propertyType`, `Status`, `id`,  `Image` FROM `properties`;");
```
Comment (Sen-i) [inline src/Hydrators/PropertyHydrator.php]: Field name don't match table.
also I believe that you should link the status and types tables to the properties table and use the respective field names.

**Review by Sen-i — CHANGES_REQUESTED** [review]

**Review by amandahrdmn — APPROVED** [review]

**Pull request merged**